qualtran/resource_counting/symbolic_counting_utils.py

```diff
@@ -24,6 +25,14 @@
 document(SymbolicFloat, """A floating point value or a sympy expression.""")
 
 
+@frozen
+class Shaped:
+    shape: tuple[SymbolicInt, ...] = field(validator=validators.instance_of(tuple))
```

Comment:
Could you write a motivating docstring? This is a qualtran-specific concept that needs to be thoroughly documented: what it is, why we need it, how you might use it

Reply:
I attempted to explain it a bit, could you check how it looks?